lib/bundler/installer.rb

```diff
@@ -75,6 +75,10 @@ def run(options)
         return
       end
 
+
+      # EXPERIMENT: Evaluate benefit if we skip resolution by comparing Gemfile checksum against the checksum saved in the lockfile.
+      return if Definition.build(Bundler.default_gemfile, Bundler.default_lockfile, nil).checksum_matches?
```

Comment:
this will also skip installation even if none of the gems are actually installed.... I think maybe you might've meant something like the following?

```diff
diff --git a/lib/bundler/installer.rb b/lib/bundler/installer.rb
index cbfdddeaf..6474d182f 100644
--- a/lib/bundler/installer.rb
+++ b/lib/bundler/installer.rb
@@ -212,6 +212,8 @@ module Bundler
     end
 
     def resolve_if_need(options)
+      return if @definition.nothing_changed?
+
       if !options[“update”] && !options[:inline] && Bundler.default_lockfile.file?
         local = Bundler.ui.silence do
           begin
```